### ISSUE TITLE
[BUG]: Etsy Dimension update issue

### ISSUE THREAD

**Issue by RatndeepGithub** [etsy/open-api#448]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When using updateListing call the dimensions are not visible on etsy seller product edit section even if the data is returned in api response .
As per the response it says that the dimensions are there  but when you see on etsy product edit page they are blank.
Any help would be appreciated.

Thanks

{"listing_id":1205197284,"user_id":591701502,"shop_id":33975309,"title":"Etsy Product For Dimension Issue","description":"lorem ipsum dolro sit amet","state":"draft","creation_timestamp":1650523595,"ending_timestamp":1661064395,"original_creation_timestamp":1650523595,"last_modified_timestamp":1650524846,"state_timestamp":1650523595,"quantity":1,"shop_section_id":37380807,"featured_rank":-1,"url":"https:\/\/www.etsy.com\/listing\/1205197284\/etsy-product-for-dimension-issue","num_favorers":0,"non_taxable":false,"is_taxable":true,"is_customizable":false,"is_personalizable":false,"personalization_is_required":false,"personalization_char_count_max":null,"personalization_instructions":null,"listing_type":"physical","tags":[],"materials":[],"shipping_profile_id":166662334988,"processing_min":1,"processing_max":3,"who_made":"i_did","when_made":"made_to_order","is_supply":false,"item_weight":1.1,"item_weight_unit":"kg","item_length":110,"item_width":110,"item_height":110,"item_dimensions_unit":"cm","is_private":false,"style":[],"file_data":"","has_variations":false,"should_auto_renew":false,"language":"en-US","price":{"amount":1000,"divisor":100,"currency_code":"USD"},"taxonomy_id":66,"production_partners":[],"skus":[]}

### Steps to reproduce

Try to use updateListing api call and update the dimension and weight of the product.

### Expected behavior

The dimensions and weight should be visible on etsy

**Comment by etsyachristensen** [comment]
@RatndeepGithub I just logged into the shop for that listing and I am not able to get to the listing edit page because you haven't completed the billing information on the account. How are you getting to that page to see the dimensions not getting updated?

**Comment by RatndeepGithub** [comment]
@etsyachristensen the shop is not live yet but I can see the dimension not being updated in etsy backend product section.
I have attached an image for reference.
![screenshot-www etsy com-2022 04 22-10_30_25](https://user-images.githubusercontent.com/49195083/164606541-a2e14543-d9c2-4d03-b2ac-916838874120.png)
Let me know if it helps .

I can share the live if the shop not being live is the reason for the dimension being not visible.

Thanks

**Comment by etsyachristensen** [comment]
When you say you are viewing the "backend product section" what is the URL? Is this on Etsy's site? I am not able to get there. I need to be able to see this for myself before I can begin debugging it.

**Comment by RatndeepGithub** [comment]
![screenshot-www etsy com-2022 04 22-20_29_58](https://user-images.githubusercontent.com/49195083/164741091-2e08ce64-4ab5-4f4a-af23-3d3732595a66.png)

**Comment by etsyachristensen** [comment]
Ok thanks. We're looking into this now. Out of curiosity, are you able to see the dimensions and weight if you enter them into a listing in the UI and save? Can you test with a different listing so that we can debug with the one you provided?

**Comment by etsyachristensen** [comment]
Ok so we have some answers. The live shop example you sent won't show the dimension fields at all in the UI because we only show the weight and dimensions on shops in US + CA because they are the only countries where we do calculated shipping.

For the former shop for listing 1205197284 you will need to complete the setup of the shop configuration and see if that data appears. We believe the information isn't showing (though it is saving) because the shop has no billing information and isn't complete.

**Comment by RatndeepGithub** [comment]
> Ok thanks. We're looking into this now. Out of curiosity, are you able to see the dimensions and weight if you enter them into a listing in the UI and save? Can you test with a different listing so that we can debug with the one you provided?

Yes , the dimensions and weight gets saved when entering them directly in Etsy UI but they are getting wiped off once the listing is updated via API.
https://www.etsy.com/your/shops/GoToStar/onboarding/listings/1205197284

I will try the same with the live listing and update the results here.

Thanks !

**Comment by hyyyper** [comment]
@etsyachristensen 
I have some additional details for this issue.

Updating dimensions via the API requires the same units that appear in the UI.

Updating a shop in US, the dimensions will only appear in the UI when the dimensions and weight units are [in, oz/lb].
Updating a shop in CA, the dimensions will only appear in the UI when the dimensions and weight units are [cm, kg].

This is a problem because when a seller successfully updates a product via the API and then tries to edit the product with the UI, the UI throws a field validation error and they have to manually fill out the dimensions and weight.

**Comment by etsyachristensen** [comment]
@hyyyper and @RatndeepGithub Are you still having this issue? We are closing the issue board soon so we need to either close this or move it to discussions. Sorry it has been so long.
